src/Spago/Glob.purs

```diff
@@ -105,11 +105,10 @@ fsWalk cwd ignorePatterns includePatterns = Aff.makeAff \cb -> do
     updateIgnoreMatcherWithGitignore :: Entry -> Effect Unit
     updateIgnoreMatcherWithGitignore entry = do
       let
-        gitignorePath = entry.path
         -- directory of this .gitignore relative to the directory being globbed
-        base = Path.relative cwd (Path.dirname gitignorePath)
+        base = Path.dirname entry.path `Path.relativeTo` root
 
-      try (SyncFS.readTextFile UTF8 entry.path) >>= traverse_ \gitignore -> do
+      try (FS.readTextFileSync entry.path) >>= traverse_ \gitignore -> do
         let
           gitignored = testGlob <$> (splitGlob $ gitignoreFileToGlob base gitignore)
```

Comment:
I found this slightly concerning place here. We're checking globs against other globs in an effort to verify if this gitignore conflicts with any of the patterns in parameters, and this doesn't always work as you would expect it to.

```
> mm = require('micromatch')
> mm.matcher("**/foo/bar")("**/foo/bar")
true
> mm.matcher("**/foo/bar")("**/foo/*")
false
```

Obviously `"**/foo/*"` overlaps `"**/foo/bar"`, but the matcher can't see it, because it expects the second parameter to be a file name, not a pattern.